core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -759,10 +759,16 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
 
       val future = result.lowWatermarks().get(topicPartition)
       try {
-        lowWatermark = future.get(1000L, TimeUnit.MILLISECONDS).lowWatermark()
+        lowWatermark = future.get().lowWatermark()
         lowWatermark == 5L
       } catch {
-        case e: LeaderNotAvailableException => false
+        case e: TimeoutException => false
+        case e: ExecutionException => {
```

Comment:
You can do this as:

```scala
case e: ExecutionException if e.getCause == LeaderNotAvailableException => false
```

Reply:
I've refactored as per your suggestion, however the == didn't work as comparison for me since "LeaderNotAvailableException is not a value", so I changed it to isInstanceOf[].